### PR TITLE
modified the various (ES)producers for express stream and HLT

### PR DESCRIPTION
**Pull request by gennai**
I have modified the Online BS Producers to take the values from the transient record produced in the related ESProducer.
This is intended for HLT and express stream reconstruction for Run3. There is a new parameter that if set to False, it would revert back to Run2 code.

A second PR on a ERA modifier configuration will follow.

#### PR validation:

I have checked the behavior of the code using the test_scalers.py script modified accordingly

### CODE REVIEW

**Comment by gennai** [comment]
@mtosi @mmusich @francescobrivio you may want to check this one

File: RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc [hunk 1]
```diff
@@ -48,94 +56,144 @@ BeamSpotOnlineProducer::BeamSpotOnlineProducer(const ParameterSet& iconf)
...
+
+      if (spotDB.GetBeamType() != 2){            
+       if (shoutMODE) {
+          edm::LogWarning("BeamSpotFromDB")
+              << "Online Beam Spot producer falls back to DB value because the ESProducer returned a fake beamspot ";
+        }
+        fallBackToDB = true;
+      }else{
+        // translate from BeamSpotObjects to reco::BeamSpot
+        reco::BeamSpot::Point apoint(spotDB.GetX(), spotDB.GetY(), spotDB.GetZ());
```
Comment (mmusich) [inline RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:109]: shouldn't also this mode support the frame conversion?
See:
https://github.com/cms-sw/cmssw/blob/9519f0955cafa3e479b4d085587782d316c5a8cb/RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc#L91-L93

File: RecoVertex/BeamSpotProducer/test/test_scalars.py [hunk 2]
```diff
@@ -70,20 +17,30 @@
...
+
+                                            ),
+                                            cms.PSet(
+                                                record = cms.string('BeamSpotOnlineHLTObjectsRcd'),
+                                                tag = cms.string('BeamSpotOnlineTestHLT'),
+                                                refreshTime = cms.uint64(1)
+                                               )
 
+                                ),
+                                        connect = cms.string('oracle://cms_orcon_prod/CMS_CONDITIONS')
```
Comment (mmusich) [inline RecoVertex/BeamSpotProducer/test/test_scalars.py:41]: this in general won't work unless you have the appropriate credentials. 
For a test can't you use `frontier` instead?

File: RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc (hunk 1 above)
Comment (gennai) [reply RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:109]: I was unsure about this, if you think it is the proper thing to do I can do that. I did not know what this changeFrame was used for

Comment (mmusich) [reply RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:109]: well, if you don't do it, it won't reproduce the previous behaviour, so I think it should be added.

Comment (gennai) [reply RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:109]: done

File: RecoVertex/BeamSpotProducer/test/test_scalars.py (hunk 2 above)
Comment (gennai) [reply RecoVertex/BeamSpotProducer/test/test_scalars.py:41]: checking now

**Comment by mmusich** [comment]
code-checks

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
test parameters:
 - workflow = 138.2

**Comment by mmusich** [comment]
@cmsbuild, please test with #34299

**Review by jpata — COMMENTED** [review]
I checked the workflow 138.2 and the only reco change is that the beamspot type is now `2` (it was `-1` previously). As I understand this is thanks to #34299, so this new code path is now also tested.

**Comment by qliphy** [comment]
test parameters:

    workflow = 138.2

**Comment by qliphy** [comment]
please test

**Comment by francescobrivio** [comment]
+alca

**Comment by jpata** [comment]
+reconstruction
- adds a new `useTransientRecord` option to `BeamSpotOnlineProducer` for HLT and Run3 express stream, off by default
- no reco changes, except in the cosmics workflow 138.2, the beamspot type is now 2 (it was -1 previously)

**Comment by qliphy** [comment]
+1

**Pull request merged**